dvc/remote/index.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import pathlib
+import pickle
+
+logger = logging.getLogger(__name__)
+
+
+class RemoteIndex(object):
+    """Class for locally indexing remote checksums.
+
+    Args:
+        repo: repo for this remote
+    """
+
+    INDEX_SUFFIX = ".idx"
+
+    def __init__(self, repo, name):
+        if repo and hasattr(repo, "index_dir") and name:
```

Comment:
I guess some tests were failing?

Reply:
Yeah, there are some remote specific unit/func tests that pass `None` or mocked objects in place of a valid repo

Reply:
@pmrowla Ideally it would be great to fix those tests to mock it properly, but I'm fine with this approach for now. IIRC we have a similar piece of code elsewhere (probably not even one). :+1: